pallets/eth-bridge/src/lib.rs

```diff
@@ -544,6 +561,46 @@ pub mod pallet {
                 }
             }
         }
+
+        fn on_idle(_block: BlockNumberFor<T>, mut remaining_weight: Weight) -> Weight {
```

Comment:
I am not sure if we want to de-prioritise Ethereum events by using onIdle. IMO we want to process them as quickly as possible.

Reply:
We can't process all at once to avoid exceed block limits.
First fraction is processed at once. the rest are added on on_idle.